File: query_mongo.js
import mongoose from 'mongoose';
import dotenv from 'dotenv';
import fs from 'fs';

dotenv.config({ path: '../CSRL-APP-backed/.env' });

const uri = process.env.MONGODB_URI;

async function run() {
  await mongoose.connect(uri);
  const db = mongoose.connection.db;
  
  const rawMarks = await db.collection('studentrawmarks').find({ studentId: '2601001', testId: 'FMT02' }).toArray();
  const weakTopics = await db.collection('studentweaktopics').find({ studentId: '2601001', testId: 'FMT02' }).toArray();
  
  fs.writeFileSync('debug_output.json', JSON.stringify({ rawMarks, weakTopics }, null, 2));
  console.log("Saved to debug_output.json");
  process.exit(0);
}
run();

File: query_mongo.py
import os
from pymongo import MongoClient
import json
from bson import json_util

uri = "mongodb+srv://csrl:csrl2024@csrl-cluster.a8l72.mongodb.net/?retryWrites=true&w=majority&appName=csrl-cluster"
client = MongoClient(uri)
db = client['test'] # or whichever db is default

raw = list(db.studentrawmarks.find({"studentId": "2601001", "testId": "FMT02"}))
weak = list(db.studentweaktopics.find({"studentId": "2601001", "testId": "FMT02"}))

with open("debug_output.json", "w") as f:
    f.write(json_util.dumps({"rawMarks": raw, "weakTopics": weak}, indent=2))
print("Saved to debug_output.json")
